clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -691,10 +691,12 @@ public Node provide() {
         }
 
         Call(String callName, long deadlineMs, NodeProvider nodeProvider,
-            Optional<Integer> numTries, Optional<Long> nextAllowedTryMs) {
+            int numTries, long nextAllowedTryMs) {
             this(false, callName, deadlineMs, nodeProvider);
-            this.tries = numTries.orElse(0);
-            this.nextAllowedTryMs = nextAllowedTryMs.orElse(0L);
+            assert numTries >= 0;
+            assert nextAllowedTryMs >= 0;
```

Comment:
I am not sure we enable java asserts when running Kafka server. Lets check the condition and throw `IllegalArgumentException` instead.

Reply:
Done.